### PR TITLE
[main] Updating `python-requests` dependencies after a version upgrade.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

A recent update of `python-requests` changed its dependency from `python-chardet` to `python-charset-normalized`. While it had a fallback mechanism to still work with the older dependency, it seems it would still fail in a few cases. Switching the package to build with the newer dependency resolved the issue.

###### Change Log  

- Switched `python-requests` to build against `python-charset-normalizer`.
- Added `python-charset-normalizer` and its dependencies.
- Added `pyproject-rpm-macros` to the worker chroot to enable parsing of the new specs.
- Updated `findutils` to provide `/usr/bin/find`.

###### Does this affect the toolchain?  

Yes.

###### Associated issues  

- #2216 

###### Test Methodology

- Local package builds.

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
Changes look good to me.
A few clarifications requested though

File: SPECS/findutils/findutils.spec [hunk 1]
```diff
@@ -10,6 +10,9 @@ Distribution:   Mariner
 Source0:        https://ftp.gnu.org/gnu/findutils/%{name}-%{version}.tar.xz
 Conflicts:      toybox
 
+# Required to unblock automatic BR resolution for some Python packages.
+Provides:       /usr/bin/find
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/findutils/findutils.spec:14]: This is the package for `find`.;) It's just that the scripts generating the BRs for one of the new Python packages ends up asking for `/usr/bin/find`, which is still a valid path (it's a link to `/bin`) but it doesn't  show up as an implicit provide from `findutils`.

File: SPECS/python-wheel/python-wheel.spec [hunk 2]
```diff
@@ -91,11 +83,9 @@ install -p dist/%{python_wheelname} -t %{buildroot}%{python_wheeldir}
 %endif
 
 
-%if %{with tests}
 %check
 rm setup.cfg
 PYTHONPATH=%{buildroot}%{python3_sitelib} py.test-3 -v --ignore build
```
[the remark this answers is not in the document]
Comment (PawelWMS) [reply SPECS/python-wheel/python-wheel.spec:88]: Haven't checked to be honest - this is just a migration from `SPECS-EXTENDED` and I didn't want to bloat the PR even more with test fixes.

File: SPECS/findutils/findutils.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfrw) [reply SPECS/findutils/findutils.spec:14]: TIL .. all along i was thinking find is a `coreutil`.
I stand corrected :) Thanks

**Pull request merged**